x/pairing/keeper/unresponsive_provider.go

```diff
@@ -227,14 +228,32 @@ func (k Keeper) getCurrentProviderStakeStorageList(ctx sdk.Context) []epochstora
 
 // Function that punishes providers. Current punishment is freeze
 func (k Keeper) punishUnresponsiveProvider(ctx sdk.Context, epochs []uint64, provider, chainID string, complaintCU uint64, servicedCU uint64, providerEpochCuMap map[uint64]types.ProviderEpochComplainerCu) error {
-	// freeze the unresponsive provider
-	err := k.FreezeProvider(ctx, provider, []string{chainID}, "unresponsiveness")
-	if err != nil {
-		utils.LavaFormatError("unable to freeze provider entry due to unresponsiveness", err,
-			utils.Attribute{Key: "provider", Value: provider},
-			utils.Attribute{Key: "chainID", Value: chainID},
+	if !utils.IsBech32Address(provider) {
+		return utils.LavaFormatWarning("Freeze_get_provider_address", fmt.Errorf("invalid address"),
+			utils.Attribute{Key: "providerAddress", Value: provider},
 		)
 	}
+
+	stakeEntry, found := k.epochStorageKeeper.GetStakeEntryByAddressCurrent(ctx, chainID, provider)
```

Comment:
punishUnresponsiveProvider is called by PunishUnresponsiveProviders which already fetches all the stake entries. You can change the code there so it'll keep the whole stake entries instead of just the provider address and get the relevant stake entry as an argument of punishUnresponsiveProvider

Do this to avoid the costly GetStakeEntryByAddressCurrent

Reply:
https://github.com/lavanet/lava/pull/1443/commits/f1a0a0463a848c0605e28e3b068220f5afdf8ccb